src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -184,14 +185,32 @@ int SysInfo::getCpuCores() const
 int SysInfo::getCpuMHz() const
 {
     int retVal { 0 };
-    std::map<std::string, std::string> systemInfo;
-    getSystemInfo(WM_SYS_CPU_DIR, ":", systemInfo);
+    int cpuFreq { 0 };
+    const auto cpusInfo { Utils::enumerateDir(WM_SYS_CPU_FREC_DIR) };
+    constexpr auto NAME_DIRECCTORY_CPU_FREQ {"^cpu[0-9]+"};
+    const std::regex rex{NAME_DIRECCTORY_CPU_FREQ};
 
-    const auto& it { systemInfo.find("cpu MHz") };
-
-    if (it != systemInfo.end())
+    for (const auto cpu : cpusInfo)
     {
-        retVal = std::stoi(it->second) + 1;
+        std::smatch sm;
+
+        if (regex_search(cpu, sm, rex))
```

Comment:
Using std::regex_match avoid using extraction mechanism and start-of-string matching pattern in the regex ('^') 
```suggestion
        if (std::regex_match(cpu, cpuDirectoryRegex))
```